src/fragments/lib/push-notifications/android/record_notifications/10_handle_notification_received.mdx

```diff
@@ -0,0 +1,104 @@
+<Callout>
+
+These APIs do not need to be called unless you choose to not use our service extension or you want to support multiple push providers.
+
+</Callout>
+
+## Handle Notification Received
+
+You can display the notification in the system tray if the app is in a background or killed state by calling `Amplify.Notifications.Push.handleNotificationReceived()` from the Messaging Service class. It also calls `recordNotificationReceived` irrespective of the app state.
+
+<BlockSwitcher>
+<Block name="Java">
+
+```java
+class MyAppService extends FirebaseMessagingService() {
+    @Override
+    public void handleIntent(Intent intent) {
+        Bundle data = intent != null ? intent.getExtras() : new Bundle()
+        // First remove any parameters that shouldn't be passed to the app
+        // * The wakelock ID set by the WakefulBroadcastReceiver
+        data.remove("androidx.content.wakelockid")
+        // Convert the RemoteMessage into a PinpointNotificationPayload
+        PinpointNotificationPayload notificationPayload = PinpointNotificationPayload.createFromRemoteMessage(new RemoteMessage(data));
+
+        // let Amplify handle foreground and background message received
+        Amplify.Notifications.Push.handleNotificationReceived(notificationPayload,
```

Comment:
Need a null check for notificationPayload before calling this API. If the notification is not sent from pinpoint, notificationPayload will be null. We need to document this behavior.